src/shared/PrivateRoute.jsx

```diff
@@ -1,22 +1,21 @@
-/* eslint-disable react/jsx-props-no-spreading */
 import React, { useContext } from "react";
 import { Route, Redirect } from "react-router-dom";
 import PropTypes from "prop-types";
-import { Container } from "@material-ui/core";
-import { AuthContext } from "./context/auth";
 
-function PrivateRoute({ component: Component }) {
+import { AuthContext } from "../context/auth";
+
+function PrivateRoute({ component: Component, ...rest }) {
   const { user } = useContext(AuthContext);
 
   return (
     <Route
-      render={(props) =>
+      // eslint-disable-next-line react/jsx-props-no-spreading
+      {...rest}
+      render={({ location }) =>
         user ? (
-          <Container>
-            <Component {...props} />
-          </Container>
+          <Component />
         ) : (
-          <Redirect to="/login" />
+          <Redirect to={{ pathname: "/login", state: { referer: location } }} />
```

Comment:
For my own understanding, is `referer` the page that the user is coming from?

Reply:
yes! if you're interested you can find the docs about it [here](https://developer.mozilla.org/en-US/docs/Web/API/Document/referrer) 😄